src/WebJobs.Script.Analyzers/AssemblyCache.cs

```diff
@@ -0,0 +1,190 @@
+                _mapRef["mscorlib"] = typeof(object).Assembly;
+                _mapRef[WebJobsAssemblyName] = typeof(Microsoft.Azure.WebJobs.FunctionNameAttribute).Assembly;
+                _mapRef[WebJobsHostAssemblyName] = typeof(Microsoft.Azure.WebJobs.JobHost).Assembly;
+
+                // JSON.Net?
+            }
+
+            // Produce tooling object
+            var webjobsStartups = new List<Type>();
+            foreach (var path in _map.Values)
+            {
+                // We don't want to load and reflect over every dll.
+                // By convention, restrict based on filenames.
+                var filename = Path.GetFileName(path);
+                // TODO: I assume this is so general to avoid missing custom extensions, but can it be tightened up for performance?
+                if (!filename.ToLowerInvariant().Contains("extension"))
+                {
+                    continue;
+                }
+                if (path.Contains(@"\ref\"))    // Skip reference assemblies.
+                {
+                    continue; 
+                }
+
+                Assembly assembly;
+                try
+                {
+                    // See GetNuGetPackagesPath for details
+                    // Script runtime is already setup with assembly resolution hooks, so use LoadFrom
+                    assembly = Assembly.LoadFrom(path);
+
+                    string asmName = new AssemblyName(assembly.FullName).Name;
+                    _mapRef[asmName] = assembly;
+
+                    var test = assembly.GetCustomAttributes<WebJobsStartupAttribute>().Select(a => a.WebJobsStartupType);
+                    if (test.Count() > 0)
+                    {
+                        webjobsStartups.AddRange(test);
+                    }
+                }
+                catch (Exception e)
+                {
+                    // Could be a reference assembly.
+                    continue;
+                }
+            }
+
+            var host = new HostBuilder()
+                .ConfigureWebJobs(b =>
+                {
+                    b.AddAzureStorageCoreServices()
+                        .UseExternalStartup(new CompilationWebJobsStartupTypeLocator(_mapRef.Values.ToArray()));
+                })
+                .Build();
+            var tooling = (JobHostMetadataProvider)host.Services.GetRequiredService<IJobHostMetadataProvider>();
+
+            lock (this)
+            {
+                this._projectCount = count;
+                this._tooling = tooling;
+            }
+        }
+
+        public void Register()
+        {
+            if (_registered)
+            {
+                return;
+            }
+            AppDomain.CurrentDomain.AssemblyResolve += CurrentDomain_AssemblyResolve;
+            _registered = true;
+        }
+
+        private Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
+        {
+            var an = new AssemblyName(args.Name);
```

Comment:
nit: I think it might be easier to follow the code with the full variable name
```suggestion
            var assemblyName = new AssemblyName(args.Name);
```

Reply:
Agreed here; changing.